src/foam/box/SessionServerBox.java

```diff
@@ -67,29 +70,27 @@ public void send(Message msg) {
 
         session.touch();
 
+        // If this service has been configured to require authentication, then
+        // throw an error if there's no user in the context.
         if ( authenticate_ && session.getUserId() == 0 ) {
           msg.replyWithException(new AuthenticationException());
           return;
         }
 
         if ( user != null ) {
-          Group group = (Group) groupDAO.find(user.getGroup());
+          Group group = (Group) x.get("group");
```

Comment:
A side-effect of this is that if we change permissions in our system they will no longer be reflected in logged in users. Users who are already logged in will continue to have those permissions until they logout.

Reply:
I think this issue is significant.

Reply:
I suppose if we really needed to revoke privileges we have the ability to manually terminate sessions for users in the affected group.  And thus far not something that's come up.

Reply:
I actually came across this issue while working on the signing officer stuff. @lchanmann pointed it out in a review he did. My solution was to loop through the sessions where the user id (and optionally agent id) matched the user whose permissions we were trying to revoke/change and update the user's group in the context in the session and put it back to the sessionDAO.

See commit https://github.com/nanoPayinc/NANOPAY/pull/6320/commits/373ffea4f8887e65974ecf6187157a154f816060

That only works in a specific situation though, it obviously doesn't cover all potential cases where this could happen.